### DISCUSSION TITLE
How to run Job before application deploy

### DISCUSSION THREAD

**Discussion by h3poteto** [fluxcd/flux2#448] [Q&A] [answered]
In my case, I want to run a Job before flux2 apply deployment, because I have to execute DB migration in deploy. For example `rails db:migrate` .
Does flux2 support such a function? I hope flux2 can easily control the apply timing of each resources.

**Comment by stefanprodan** [comment, marked as answer]
We have no support for pre/post deploy hooks as we don't plan to recreate Helm. You can use an init container inside your app deployment and run the migration that way.

**Reply by h3poteto** [reply]
Thank you for your quick response.
I see, I've thought about init container, but pre/post deploy hooks are more smarter.

Do you have any plans to support this function?

**Reply by stefanprodan** [reply]
We already support pre/post hooks when you package your app with Helm and deploy it with `HelmReleases`, by default helm-controller runs the hooks on install and upgrade.

**Reply by h3poteto** [reply]
Is the feature documented? I can't find it.

**Reply by stefanprodan** [reply]
https://helm.sh/docs/topics/charts_hooks/

**Reply by h3poteto** [reply]
I see, I understand, the feature is provided by helm. So I can't use it when I use `Kustomization`.
